src/main/java/WayofTime/bloodmagic/block/BlockDimensionalPortal.java

```diff
@@ -76,27 +76,24 @@ public void onEntityCollision(World world, BlockPos pos, IBlockState blockState,
         if (!world.isRemote && world.getTileEntity(pos) instanceof TileDimensionalPortal) {
             TileDimensionalPortal tile = (TileDimensionalPortal) world.getTileEntity(pos);
 
-            if (LocationsHandler.getLocationsHandler() != null) {
-                ArrayList<PortalLocation> linkedLocations = LocationsHandler.getLocationsHandler().getLinkedLocations(tile.portalID);
-
-                if (linkedLocations != null && !linkedLocations.isEmpty() && linkedLocations.size() > 1) {
-                    if (world.getTileEntity(tile.getMasterStonePos()) != null && world.getTileEntity(tile.getMasterStonePos()) instanceof IMasterRitualStone) {
-                        IMasterRitualStone masterRitualStone = (IMasterRitualStone) world.getTileEntity(tile.getMasterStonePos());
-                        if (linkedLocations.get(0).equals(new PortalLocation(masterRitualStone.getBlockPos().up(), world.provider.getDimension()))) {
-                            PortalLocation portal = linkedLocations.get(1);
-                            if (portal.getDimension() == world.provider.getDimension()) {
-                                TeleportQueue.getInstance().addITeleport(new Teleports.TeleportSameDim(portal.getX(), portal.getY(), portal.getZ(), entity, masterRitualStone.getOwner(), false));
-                            } else {
-                                TeleportQueue.getInstance().addITeleport(new Teleports.TeleportToDim(portal.getX(), portal.getY(), portal.getZ(), entity, masterRitualStone.getOwner(), world, portal.getDimension(), false));
-                            }
-                        } else if (linkedLocations.get(1).equals(new PortalLocation(masterRitualStone.getBlockPos().up(), world.provider.getDimension()))) {
-                            PortalLocation portal = linkedLocations.get(0);
-                            if (portal.getDimension() == world.provider.getDimension()) {
-                                TeleportQueue.getInstance().addITeleport(new Teleports.TeleportSameDim(portal.getX(), portal.getY(), portal.getZ(), entity, masterRitualStone.getOwner(), false));
-                            } else {
-                                TeleportQueue.getInstance().addITeleport(new Teleports.TeleportToDim(portal.getX(), portal.getY(), portal.getZ(), entity, masterRitualStone.getOwner(), world, portal.getDimension(), false));
-                            }
-                        }
+            LocationsHandler.verifyIsInitialized();
+            ArrayList<PortalLocation> linkedLocations = LocationsHandler.getLinkedLocations(tile.portalID);
+
+            if (linkedLocations != null && linkedLocations.size() > 1) {
+                if (world.getTileEntity(tile.getMasterStonePos()) != null && world.getTileEntity(tile.getMasterStonePos()) instanceof IMasterRitualStone) {
```

Comment:
`instanceof` implies nonnull. The tile obtained here should also be stored to avoid unnecessary lookups